cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/type/DatasetTypeManager.java

```diff
@@ -412,10 +416,22 @@ public Boolean call() throws DatasetModuleConflictException, IOException {
           }
 
           datasetTypeMDS.deleteModule(datasetModuleId);
-          // Also delete module jar
-          Location moduleJarLocation = locationFactory.create(module.getJarLocation());
-          if (!moduleJarLocation.delete()) {
-            LOG.debug("Could not delete dataset module archive");
+          try {
+            // Also delete module jar
+            impersonator.doAs(datasetModuleId.getNamespace().toEntityId(), new Callable<Void>() {
+              @Override
+              public Void call() throws Exception {
+                Location moduleJarLocation = locationFactory.create(module.getJarLocation());
+                if (!moduleJarLocation.delete()) {
```

Comment:
@anwar6953 do we know for sure this operation can be safely performed outside of the doAs?